### PR TITLE
flare-floss: 1.7.0 -> 2.0.0

### PR DESCRIPTION
**Pull request by sbruder**
###### Description of changes

This updates flare-floss as part of the [Summer of Nix](https://summer.nixos.org/). For this it packages the new python module `python-flirt`, updates the module `viv-utils` to a newer version and adds flirt support to it.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/flare-floss/default.nix [hunk 1]
```diff
@@ -1,45 +1,110 @@
...
+    version = "2.5.1";
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-EJzVhcrEEpf3EQPDxCrG73N58peI61TLdRvlpmO7I1o=";
+    };
+
+    propagatedBuildInputs = [ decorator_4_4_2 setuptools ];
+    checkInputs = [ nose pytest pytestCheckHook ];
+  };
```
Comment (gytis-ivaskevicius) [inline pkgs/tools/security/flare-floss/default.nix:46]: Consider defining global derivation version alias. Tho I am not sure what the correct way of doing this

**Review by gytis-ivaskevicius — APPROVED** [review]

File: pkgs/development/python-modules/viv-utils/default.nix [hunk 2]
```diff
@@ -1,45 +1,78 @@
...
     repo = "viv-utils";
     rev = "v${version}";
-    sha256 = "wZWp6PMn1to/jP6lzlY/x0IhS/0w0Ys7AdklNQ+Vmyc=";
+    sha256 = "sha256-JDu+1n1wP2Vsp2V/bKdE+RFp6bE8RNmimi4wdsatwME=";
   };
 
-  # argparse is provided by Python itself
-  preBuild = ''
-    sed '/"argparse",/d' -i setup.py
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/viv-utils/default.nix:31]: Please use a sed for this or replace all == with >=

File: pkgs/development/python-modules/viv-utils/default.nix [hunk 3]
```diff
@@ -1,45 +1,78 @@
...
-  # no tests
-  doCheck = false;
-
-  pythonImportsCheck = [
-    "viv_utils"
+  checkInputs = [
+    black
+    isort
+    mypy
+    pycodestyle
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/viv-utils/default.nix:60]:
```suggestion
```
we never want to check style

File: pkgs/development/python-modules/viv-utils/default.nix [hunk 4]
```diff
@@ -1,45 +1,78 @@
...
+    pytest
+    pytest-instafail
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/viv-utils/default.nix:62]:
```suggestion
```
instant failing tests are annoying if there are multiple

File: pkgs/development/python-modules/viv-utils/default.nix [hunk 5]
```diff
@@ -1,45 +1,78 @@
...
+    pytest-sugar
+    pytestCheckHook
+    types-setuptools
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/viv-utils/default.nix:65]:
```suggestion
```

File: pkgs/tools/security/flare-floss/default.nix [hunk 6]
```diff
@@ -1,45 +1,110 @@
...
+, pytest-cov
+, pytest-instafail
+, pytest-sugar
+, pyyaml
+, types-pyyaml
+, types-tabulate
 }:
+let
+  # required for networkx 2.5.1
+  decorator_4_4_2 = buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:24]: Please use overridePythonAttrs

File: pkgs/tools/security/flare-floss/default.nix [hunk 7]
```diff
@@ -1,45 +1,110 @@
...
+    pname = "networkx";
+    version = "2.5.1";
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-EJzVhcrEEpf3EQPDxCrG73N58peI61TLdRvlpmO7I1o=";
+    };
+
+    propagatedBuildInputs = [ decorator_4_4_2 setuptools ];
+    checkInputs = [ nose pytest pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:45]:
```suggestion
    checkInputs = [ nose pytestCheckHook ];
```

File: pkgs/tools/security/flare-floss/default.nix [hunk 8]
```diff
@@ -1,45 +1,110 @@
...
+    fetchSubmodules = true; # for tests
   };
 
-  propagatedBuildInputs = with python3.pkgs; [
-    pyyaml
-    simplejson
+  postPatch = ''
+    substituteInPlace setup.py \
+        --replace "halo==0.0.31" "halo" \
+        --replace "pydantic==1.9.1" "pydantic" \
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:63]: Also just replace all == with >=

File: pkgs/tools/security/flare-floss/default.nix [hunk 9]
```diff
@@ -1,45 +1,110 @@
...
+        --replace "pytest==7.1.2" "pytest" \
+        --replace "pyyaml==6.0" "pyyaml" \
+        --replace "tabulate==0.8.9" "tabulate" \
+        --replace "tqdm==4.64.0" "tqdm" \
+        --replace "types-PyYAML==6.0.8" "types-pyyaml" \
+        --replace "types-tabulate==0.8.9" "types-tabulate" \
+        --replace "vivisect==1.0.8" "vivisect"
+
+    substituteInPlace floss/main.py \
+        --replace 'sigs_path = os.path.join(get_default_root(), "sigs")' 'sigs_path = "'"$out"'/share/flare-floss/sigs"'
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:76]:
```suggestion
      --replace 'sigs_path = os.path.join(get_default_root(), "sigs")' 'sigs_path = "'"$out"'/share/flare-floss/sigs"'
```

File: pkgs/tools/security/flare-floss/default.nix [hunk 10]
```diff
@@ -1,45 +1,110 @@
...
+    halo
+    networkx_2_5_1
+    tqdm
+  ] ++ viv-utils.optional-dependencies.flirt;
 
-  checkInputs = with python3.pkgs; [
+  checkInputs = [
+    pytest
+    pytest-cov
+    pytest-instafail
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:92]:
```suggestion
```
coverage is not interesting to us

File: pkgs/tools/security/flare-floss/default.nix [hunk 11]
```diff
@@ -1,45 +1,110 @@
...
+    pytest-sugar
     pytestCheckHook
+    pyyaml
+    types-pyyaml
+    types-tabulate
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:97]:
```suggestion
```
we are not checking types

File: pkgs/tools/security/flare-floss/default.nix [hunk 12]
```diff
@@ -2,44 +2,73 @@
...
+  version = "2.0.0";
 
   src = fetchFromGitHub {
-    owner = "fireeye";
+    owner = "mandiant";
     repo = "flare-floss";
     rev = "v${version}";
-    sha256 = "GMOA1+qM2A/Qw33kOTIINEvjsfqjWQWBXHNemh3IK8w=";
+    sha256 = "sha256-V4OWYcISyRdjf8x93B6h2hJwRgmRmk32hr8TrgRDu8Q=";
+    fetchSubmodules = true; # for tests
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:37]:
```suggestion
    fetchSubmodules = true; # for tests
    sha256 = "sha256-V4OWYcISyRdjf8x93B6h2hJwRgmRmk32hr8TrgRDu8Q=";
```

File: pkgs/tools/security/flare-floss/default.nix [hunk 13]
```diff
@@ -2,44 +2,73 @@
 , python3
 , fetchFromGitHub
 }:
+let
+  python3' = python3.override {
```
Comment (SuperSandro2000) [inline pkgs/tools/security/flare-floss/default.nix:6]:
```suggestion
  py = python3.override {
```
we normally name it like this

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 182814` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>flare-floss</li>
<li>python310Packages.python-flirt</li>
<li>python310Packages.viv-utils</li>
<li>python39Packages.python-flirt</li>
<li>python39Packages.viv-utils</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python310Packages.viv-utils:
<details>

got build log for '/nix/store/9r20qx91pyar6al9qnzalfkfdmirq7bi-python3.10-viv-utils-0.7.5' from 'daemon'
When evaluating attribute ‘python310Packages.viv-utils’:
warning: unused-argument
Unused argument: `types-setuptools`.
Near pkgs/development/python-modules/viv-utils/default.nix:11:3:

```
   |
11 | , types-setuptools
   |   ^
```
</details>
python39Packages.viv-utils:
<details>

got build log for '/nix/store/0fir09v355par0kzq86rlsld942akj9y-python3.9-viv-utils-0.7.5' from 'daemon'
When evaluating attribute ‘python39Packages.viv-utils’:
warning: unused-argument
Unused argument: `types-setuptools`.
Near pkgs/development/python-modules/viv-utils/default.nix:11:3:

```
   |
11 | , types-setuptools
   |   ^
```
</details>
</details>

**Pull request merged**